tests/lib/rules/id-length.js

```diff
@@ -50,7 +51,10 @@ ruleTester.run("id-length", rule, {
         { code: "import something from 'y';", ecmaFeatures: { modules: true } },
         { code: "export var num = 0;", ecmaFeatures: { modules: true } },
         { code: "({ prop: obj.x.y.something }) = {};", ecmaFeatures: { destructuring: true } },
-        { code: "({ prop: obj.longName }) = {};", ecmaFeatures: { destructuring: true } }
+        { code: "({ prop: obj.longName }) = {};", ecmaFeatures: { destructuring: true } },
+        { code: "var obj = { a: 1, bc: 2 };", options: [{ "properties": "never" }] },
+        { code: "({ a: obj.x.y.z }) = {};", options: [{ "properties": "never" }], ecmaFeatures: { destructuring: true } },
+        { code: "({ prop: obj.x }) = {};", options: [{ "properties": "never" }], ecmaFeatures: { destructuring: true } }
     ],
     invalid: [
```

Comment:
Adding at least one move `invalid` case where `properties` is set to `"never"` would be good to make sure setting that does not affect anything else.